lib/fluent/plugin/out_memcached.rb

```diff
@@ -13,15 +17,22 @@ class Fluent::Plugin::MemcachedOutput < Fluent::Plugin::Output
   config_param :value_format, :string, :default => 'raw'
   config_param :param_names, :string, :default => nil # nil doesn't allowed for json
 
+  config_section :buffer do
+    config_set_default :@type, DEFAULT_BUFFER_TYPE
+    config_set_default :chunk_keys, ['tag']
+  end
+
   attr_accessor :memcached
   attr_accessor :formatter
 
   def configure(conf)
+    compat_parameters_convert(conf, :buffer)
     super
     if @value_format == 'json' and @param_names.nil?
       raise Fluent::ConfigError, "param_names MUST be specified in the case of json format"
     end
     @formatter = RecordValueFormatter.new(@increment, @value_separater, @value_format, @param_names)
+    raise Fluent::ConfigError, "'tag' in chunk_keys is required." unless @chunk_key_tag
```

Comment:
This is the internal instance variable for existence checking for tag in chunk_keys.
See: https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/output.rb#L278